src/libraries/Common/src/Interop/Linux/OpenLdap/Interop.Ldap.cs

```diff
@@ -61,141 +61,158 @@ internal enum SaslChallengeType
 
 internal static partial class Interop
 {
-    [DllImport(Libraries.OpenLdap, EntryPoint = "ldap_initialize", CharSet = CharSet.Ansi, SetLastError = true)]
-    public static extern int ldap_initialize(out IntPtr ld, string hostname);
+    internal static partial class Ldap
+    {
+        static Ldap()
+        {
+            // OpenLdap must be initialized on a single thread, once this is done it allows concurrent calls
+            // By doing so in the static constructor we guarantee this is run before any other methods are called.
+
+            // we call ldap_get_option_int to get an option and trigger the initialization as reccomended by
+            // https://www.openldap.org/software//man.cgi?query=ldap_init
+            int unused = 0;
+            ldap_get_option_int(IntPtr.Zero, LdapOption.LDAP_OPT_DEBUG_LEVEL, ref unused);
```

Comment:
Just one note here, at least in Linux native library will fail the call when the option isn't recognized, so we could either add a try catch here in case it isn't, or alternatively either confirm debug level is present on all implementations or use another option we know will like protocol version

Reply:
I chose an option that did not require a handle nor any allocation.  It appears to me that this option is present everywhere: https://github.com/openldap/openldap/blob/b06f5b0493937fc28f2cc86df1d7f464aa4504d8/libraries/libldap/options.c#L366